### ISSUE TITLE
broken with react-native 0.12 & 0.13?

### ISSUE THREAD

**Issue by bryanlarsen** [alinz/react-native-webview-bridge#31]
To reproduce:

- react-native init AwesomeProject
- replace AwesomeProject#render with `return <WebView style={{flex:1}} html="<html><body><h1>Hi</h1></body></html>" />;`
- run the project and notice it works
- install react-native-webview-bridge as per instructions
- run the project and notice the blank screen you get
- npm install react-native@0.11.4
- run the project and notice that it works

Notice that it's WebView that's broken, which is really surprising.  Using WebViewBridge with or without injecting the bridge script doesn't help.

**Comment by NoelBaron** [comment]
Same here. Considering reverting to 0.11.4 if Ali is too busy to address.

**Comment by jjshammas** [comment]
+1

**Comment by alinz** [comment]
I am extremely sorry, I've been very busy. I will address this issue ASAP. I will also accept PRs :+1:  if you guys can help me out!

**Comment by jjshammas** [comment]
@NoelBaron  @bryanlarsen 
I was able to fix with this quick change: https://github.com/jjshammas/react-native-webview-bridge/commit/7c1cc248289a668300db119d077c2d80039e76be

But I'm not using the bridge very extensively in my project, so give it a shot. If it seems to be working for you guys as well, this may be the solution.

**Comment by NoelBaron** [comment]
It solved the loading issue for me, but for some reason I cannot access WebViewBridge from the remote website's javascript. It looks like the injection fails.

**Comment by dsun412** [comment]
@NoelBaron I was able to get it working on React Native 0.13 by converting webview-bridge-script.js into a minified JS string, and then manually retrieve the handlerId using:
```
var webviewRef = this.refs[WEBVIEW_REF];
webviewRef.getWebViewBridgeHandler(function (handlerId) {
    // inject JS
});
```
Finally, you can use your the `injectedJavaScript` property on the WebView to manually inject the JS without having to touch any Objective-C code.

**Comment by bryanlarsen** [comment]
@rick-li appears to have a fix in #32

**Comment by alinz** [comment]
OK guys, thanks to @rick-li, it is now fully functional in latest react-native. One other thing is, I have moved all the files into proper xcodeproj. You just have to add it to your project and make sure you to add to your `Link Binary With Libraries` project. I will update the Readme.
